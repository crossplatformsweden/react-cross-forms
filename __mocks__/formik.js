module.exports = {
  ErrorMessage: 'div',
  Field: 'div',
  Form: 'div',
  Formik: 'div',
};
