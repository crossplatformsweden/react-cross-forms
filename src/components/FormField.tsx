import React, { InputHTMLAttributes } from 'react';
import { Field, ErrorMessage } from 'formik';

/**
 * Extends input props with `label`
 * @property label - Label displayed for the user
 * @property as - element to render as instead of `<input />`
 */
interface IFormField
  extends React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > {
  label?: string;
  as?: string;
}

/**
 * Wraps a label with formik field and error message.
 *
 * Must be wrapped in a `<Formik></Formik>`
 *
 * https://jaredpalmer.com/formik/
 * @param props Input props
 */
export const FormField: React.FunctionComponent<IFormField> = ({
  label,
  ...props
}) => (
  <div>
    <label htmlFor={props.id || props.name}>{label || props.name}</label>
    <Field {...props}>{props.children}</Field>
    <ErrorMessage name={props.name || ''} component="span" />
  </div>
);
