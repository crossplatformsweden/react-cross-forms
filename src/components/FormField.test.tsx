import React from 'react';
import renderer from 'react-test-renderer';
import { FormField } from './FormField';
import { Formik } from 'formik';

describe('components', () => {
  describe('<FormField />', () => {
    const formikProps = {
      initialValues: {},
      onSubmit: jest.fn(),
    };
    test('renders correctly', () => {
      const instance = renderer
        .create(
          <Formik {...formikProps}>
            <FormField />
          </Formik>
        )
        .toJSON();
      expect(instance).toMatchSnapshot();
    });
    test('When no props', async () => {
      renderer.create(
        <Formik {...formikProps}>
          <FormField />
        </Formik>
      );
    });
    test('When has props', async () => {
      renderer.create(
        <Formik {...formikProps}>
          <FormField type="email" name="email" />
        </Formik>
      );
    });
    test('When two <FormField/>', async () => {
      renderer.create(
        <Formik {...formikProps}>
          <FormField />
          <FormField />
        </Formik>
      );
    });
  });
});
