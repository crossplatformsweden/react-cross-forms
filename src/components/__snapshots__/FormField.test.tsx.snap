// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <FormField /> renders correctly 1`] = `
<div
  initialValues={Object {}}
  onSubmit={[MockFunction]}
>
  <div>
    <label />
    <div />
    <div
      component="span"
      name=""
    />
  </div>
</div>
`;
